### PR TITLE
Intel/CI: Update dmabuf stage to support subdevice testing

### PR DESCRIPTION
**Pull request by Juee14Desai**
This commit updates the dmabuf stage to support tile-to-tile 
testing based on hierarchy update to newer agama versions.

### CODE REVIEW

File: contrib/intel/jenkins/tests.py [hunk 1]
```diff
@@ -1062,7 +1062,7 @@ def __init__(self, jobname, buildno, testname, hw, core_prov, fabric,
                                             else 0
 
         if util_prov:
-            self.prov = f'{self.core_prov}\;{self.util_prov}'
+            self.prov = f'{self.core_prov};{self.util_prov}'
```
Comment (zachdworkin) [inline contrib/intel/jenkins/tests.py:1065]: Didn't we decide we need the '\' here? I thought it was making the command line think this was two different commands separated by the ';'?

Comment (Juee14Desai) [reply contrib/intel/jenkins/tests.py:1065]: Yes, I'll fix that but I am more inclined towards using self.core_prov since we are only using verbs.

Comment (zachdworkin) [reply contrib/intel/jenkins/tests.py]: I think its better to have it this way even if we are only testing verbs. That way if we add more things in the future we won't have to figure out the same bug again.

File: contrib/intel/jenkins/tests.py [hunk 2]
```diff
@@ -1073,27 +1073,40 @@ def __init__(self, jobname, buildno, testname, hw, core_prov, fabric,
...
+                            'H' : ['-d 0', '-d 1']
+                        },
+                'D2H'   : {
+                            '-d 0' : 'H',
+                            '-d 1' : 'H'
+                        },
+                'D2D'   : {
+                            '-d 0' : ['-d 1', '-d 2', '-d 3'],
+                            '-d 1' : ['-d 2', '-d 3']
+                        }
```
Comment (j-xiong) [inline contrib/intel/jenkins/tests.py:1090]: The control logic later would be much simplified if the test configuration are defined as command line options directly and the lists are expanded. e.g.:

```
      'H2D'  : {
              '-m malloc' : '-m device -d 0'
              '-m malloc' : '-m device -d 1'
       },
```

Comment (Juee14Desai) [reply contrib/intel/jenkins/tests.py:1090]: I figured that adding -m device and -m malloc to each config would be redundant and hence just specified H for malloc  and -d <device_no> for device. This way any combination could be added later on in the config.

Comment (j-xiong) [reply contrib/intel/jenkins/tests.py:1090]: That results in the triple loop control structure that is much harder to understand vs the single loop if the redundancy is tolerated in the configuration. IMHO, the configuration with the `-m` option added is also easier to read.

**Comment by Juee14Desai** [comment]
Intel CI failure is due to an unrelated summary stage failure, dmabuf stage has passed and summary shows correct results.

**Review by zachdworkin — APPROVED** [review]

**Pull request merged**